src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -932,14 +910,52 @@ protected override void EndProcessing()
             {
                 // Load the source if they want to load from a file
                 if (String.Equals(ParameterSetName, "FromPath", StringComparison.OrdinalIgnoreCase) ||
-                    String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase))
+                    String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase)
+                   )
                 {
-                    this.sourceCode = "";
-                    foreach (string file in paths)
+                    if (paths.Length == 1)
+                    {
+                        sourceCode = File.ReadAllText(paths[0]);
+                    }
+                    else
                     {
-                        this.sourceCode += System.IO.File.ReadAllText(file) + "\n";
+
+                        long initLength = 0;
+
+                        foreach (string file in paths)
+                        {
+                            FileInfo f = new FileInfo(file);
```

Comment:
I wonder what the cost is for obtaining file information.  I assume the file doesn't need to be opened?  Is the cost a reasonable trade off to StringBuilder automatic allocation?

Reply:
I couldn't find out how it was implemented in CoreFX. 😕 
In any case, if we want to handle a lot of files, we need to refactor the cmdlet so that we don't read all the files in one string, but compile file by file. I plan to do it later